test/always-be-playing.test.js

```diff
@@ -132,78 +126,172 @@ QUnit.test('skips over gap in Chrome due to video underflow', function() {
   }
 });
 
 QUnit.test('skips gap from video underflow', function() {
   QUnit.equal(
-    this.gapSkipper.gapFromVideoUnderflow_(videojs.createTimeRanges(), 0),
+    this.alwaysBePlaying.gapFromVideoUnderflow_(videojs.createTimeRanges(), 0),
     null,
     'returns null when buffer is empty');
   QUnit.equal(
-    this.gapSkipper.gapFromVideoUnderflow_(videojs.createTimeRanges([[0, 10]]), 13),
+    this.alwaysBePlaying.gapFromVideoUnderflow_(videojs.createTimeRanges([[0, 10]]), 13),
     null,
     'returns null when there is only a previous buffer');
   QUnit.equal(
-    this.gapSkipper.gapFromVideoUnderflow_(
+    this.alwaysBePlaying.gapFromVideoUnderflow_(
       videojs.createTimeRanges([[0, 10], [10.1, 20]]), 15),
     null,
     'returns null when gap is too far from current time');
   QUnit.equal(
-    this.gapSkipper.gapFromVideoUnderflow_(
+    this.alwaysBePlaying.gapFromVideoUnderflow_(
       videojs.createTimeRanges([[0, 10], [10.1, 20]]), 9.9),
     null,
     'returns null when gap is after current time');
   QUnit.equal(
-    this.gapSkipper.gapFromVideoUnderflow_(
+    this.alwaysBePlaying.gapFromVideoUnderflow_(
       videojs.createTimeRanges([[0, 10.1], [10.2, 20]]), 12.1),
     null,
     'returns null when time is less than or equal to 2 seconds ahead');
   QUnit.equal(
-    this.gapSkipper.gapFromVideoUnderflow_(
+    this.alwaysBePlaying.gapFromVideoUnderflow_(
       videojs.createTimeRanges([[0, 10], [10.1, 20]]), 14.1),
     null,
     'returns null when time is greater than or equal to 4 seconds ahead');
   QUnit.deepEqual(
-    this.gapSkipper.gapFromVideoUnderflow_(
+    this.alwaysBePlaying.gapFromVideoUnderflow_(
       videojs.createTimeRanges([[0, 10], [10.1, 20]]), 12.2),
     {start: 10, end: 10.1},
     'returns gap when gap is small and time is greater than 2 seconds ahead in a buffer');
   QUnit.deepEqual(
-    this.gapSkipper.gapFromVideoUnderflow_(
+    this.alwaysBePlaying.gapFromVideoUnderflow_(
       videojs.createTimeRanges([[0, 10], [10.1, 20]]), 13),
     {start: 10, end: 10.1},
     'returns gap when gap is small and time is 3 seconds ahead in a buffer');
   QUnit.deepEqual(
-    this.gapSkipper.gapFromVideoUnderflow_(
+    this.alwaysBePlaying.gapFromVideoUnderflow_(
       videojs.createTimeRanges([[0, 10], [10.1, 20]]), 13.9),
     {start: 10, end: 10.1},
     'returns gap when gap is small and time is less than 4 seconds ahead in a buffer');
   // In a case where current time is outside of the buffered range, something odd must've
   // happened, but we should still allow the player to try to continue from that spot.
   QUnit.deepEqual(
-    this.gapSkipper.gapFromVideoUnderflow_(
+    this.alwaysBePlaying.gapFromVideoUnderflow_(
       videojs.createTimeRanges([[0, 10], [10.1, 12.9]]), 13),
     {start: 10, end: 10.1},
     'returns gap even when current time is not in buffered range');
 });
+
+QUnit.test('check fell out of live window', function() {
```

Comment:
These tests would probably be less complicated if you split the side effects of the check out of the check itself.